utils/fs/memory/memory.go

```diff
@@ -39,7 +40,7 @@ func (fs *Memory) Open(filename string) (fs.File, error) {
 func (fs *Memory) OpenFile(filename string, flag int, perm os.FileMode) (fs.File, error) {
 	fullpath := fs.Join(fs.base, filename)
 	f, ok := fs.s.files[fullpath]
-	if !ok && flag&os.O_CREATE == 0 {
+	if !ok && !isCreate(flag) {
```

Comment:
you are not checking if flag *is* O_CREATE but if flags *has* O_CREATE set. I would rename all the check functions to hasFoo: hasCreate(flag), hasAppend(flag)... And I would create the negate ones also: hasNotCreate(flag), hasNotAppend(flag)...